packages/vats/test/localchain.test.js

```diff
@@ -0,0 +1,156 @@
+import { test as anyTest } from '@agoric/swingset-vat/tools/prepare-test-env-ava.js';
+import { reincarnate } from '@agoric/swingset-liveslots/tools/setup-vat-data.js';
+import { E } from '@endo/far';
+import { M, matches } from '@endo/patterns';
+import { makeDurableZone } from '@agoric/zone/durable.js';
+import { AmountMath, AssetKind, makeIssuerKit } from '@agoric/ertp';
+import { buildRootObject } from '../src/vat-bank.js';
+import { prepareLocalChainTools } from '../src/localchain.js';
+
+/** @type {import('ava').TestFn<ReturnType<makeTestContext>>} */
+const test = anyTest;
+
+const makeTestContext = async t => {
+  const { fakeVomKit } = reincarnate({ relaxDurabilityRules: false });
+  const provideBaggage = key => {
+    const root = fakeVomKit.cm.provideBaggage();
+    const zone = makeDurableZone(root);
+    return zone.mapStore(`${key} baggage`);
+  };
+  const baggage = provideBaggage('localchainTest');
+  const bankVat = E(buildRootObject)(null, null, baggage);
+
+  const zone = makeDurableZone(baggage);
+
+  /** @type {undefined | ERef<import('../src/types.js').BridgeHandler>} */
+  let bankHandler;
+
+  /** @type {import('../src/types.js').ScopedBridgeManager} */
+  const fakeBridgeManager = zone.exo('fakeBankBridgeManager', undefined, {
+    async fromBridge(obj) {
+      t.is(typeof obj, 'string');
+    },
+    async toBridge(obj) {
+      let ret;
+      switch (obj.type) {
+        case 'VLOCALCHAIN_ALLOCATE_ADDRESS': {
+          ret = 'agoricfoo';
+          break;
+        }
+        case 'VBANK_GET_BALANCE': {
+          const { address, denom, type: _type, ...rest } = obj;
+          t.is(address, 'agoricfoo');
+          t.deepEqual(rest, {});
+          if (denom === 'ubld') {
+            ret = '100000000';
```

Comment:
tracking the balance thru `VBANK_GIVE` and `VBANK_GRAB` seems cost-effective for this test

Reply:
I did not observe any event for `VBANK_GIVE` or `VBANK_GRAB`, so I've removed them. The second withdraw attempt in the test demonstrates a failure due to a lack of balance, which I think should be sufficient. 

Aside - this events are still not observed when the (mock) `bridgeManager` is passed in to `makeBankManager` on L69.